src/main/java/net/luminis/http3/impl/InterfaceBoundDatagramSocketFactory.java

```diff
@@ -0,0 +1,24 @@
+package net.luminis.http3.impl;
+
+import net.luminis.quic.DatagramSocketFactory;
+
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+
+public class InterfaceBoundDatagramSocketFactory implements DatagramSocketFactory {
+    private final InetAddress address;
+    public InterfaceBoundDatagramSocketFactory(InetAddress address) {
```

Comment:
I wonder what happens if someone passes a address that is not local. Why not add a unit test that shows what happens (e.g. which exception is thrown) in such case?